src/Equinor.Procosys.Library.Infrastructure/Caching/CacheManager.cs

```diff
@@ -18,7 +24,11 @@ public class CacheManager : ICacheManager
             var instance = Get<T>(key);
             if (instance != null)
             {
-                return instance;
+                var t = instance as Task;
+                if (t == null || t.IsCompletedSuccessfully)
```

Comment:
(instance is not Task t || t.IsCompletedSuccessfully) ? 

More readable I think. this is not super obvious anyhow. 

Rewrite is so that if you already have a  cache operation going, you don't start another one? Could consider some comments.

Reply:
pushed changes @TordJoranger ... is my comment descriptive?

Reply:
Yes :)